app/collections/show-list.jsx

```diff
@@ -0,0 +1,345 @@
+/*
+ * decaffeinate suggestions:
+ * DS102: Remove unnecessary code created because of implicit returns
+ * DS205: Consider reworking code to avoid use of IIFEs
+ * DS207: Consider shorter variations of null checks
+ * Full docs: https://github.com/decaffeinate/decaffeinate/blob/main/docs/suggestions.md
+ */
+import { Component, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import {Link} from 'react-router';
+import Translate from 'react-translate-component';
+import apiClient from 'panoptes-client/lib/api-client';
+import intersection from 'lodash/intersection';
+import pick from 'lodash/pick';
+import classNames from 'classnames';
+import counterpart from 'counterpart';
+import alert from '../lib/alert';
+import Paginator from '../talk/lib/paginator';
+import SubjectViewer from '../components/subject-viewer';
+import Loading from '../components/loading-indicator';
+import CollectionsManager from './collections-manager';
+import getSubjectLocation from '../lib/getSubjectLocation';
+
+const VALID_COLLECTION_MEMBER_SUBJECTS_PARAMS = ['page', 'page_size'];
+
+counterpart.registerTranslations('en', {
+  collectionSubjectListPage: {
+    error: 'There was an error listing this collection.',
+    noSubjects: 'No subjects in this collection.'
+  }
+});
+
+const DEFAULT_HANDLER = () => true;
+
+function SubjectNode({
+  addSelected = DEFAULT_HANDLER,
+  canCollaborate = false,
+  collection = null,
+  onDelete = DEFAULT_HANDLER,
+  projectContext = null,
+  removeSelected = DEFAULT_HANDLER,
+  selecting = false,
+  selected = false,
+  subject,
+  user = null
+}) {
+  const [isFavorite, setIsFavorite] = useState(false);
+  const [project, setProject] = useState(projectContext);
+
+  async function checkFavorite(project) {
+    if (collection?.favorite && (collection.links.owner.id === user?.id)) {
+      setIsFavorite(true);
+    } else if (user != null) {
+      const query = {
+        favorite: true,
+        project_ids: project.id,
+        owner: user.login
+      };
+      let isFavorite = false;
+
+      const [favoritesCollection] = await apiClient.type('collections').get(query);
+      if ((favoritesCollection != null) && (favoritesCollection.links.subjects != null)) {
+        isFavorite = favoritesCollection.links.subjects.includes(subject.id);
+      }
+      setIsFavorite(isFavorite);
+    }
+  }
+
+  useEffect(function () {
+    async function getSubjectProject() {
+      const project = await subject.get('project');
+      setProject(project);
+    }
+
+    if (projectContext) {
+      if (projectContext !== project) {
+        setProject(projectContext);
+      }
+    } else {
+      getSubjectProject();
+      checkFavorite(project);
```

Comment:
Should this run after the project has updated? Here, it's running before the project is fetched.